Sources/SyrupCore/Stencil Extensions/SelectionSetExtension.swift

```diff
@@ -327,28 +336,50 @@ class SelectionSetExtension: Extension {
 			return "undefined"
 		}
 		
-		let typeCondition = args.first as? String
 		let spacingAmount = args[1] as? Int ?? 0
 		let spacing = String(repeating: " ", count: spacingAmount)
 		
 		var fieldRenders: [String] = []
-		var fragmentSpreadRenders: [String] = []
-		
-		var combinedSelections: [SelectionSetVisitor.Selection] = []
 		
 		for selection in selections {
-			if selection.field != nil {
-				combinedSelections.append(selection)
-			} else if selection.inlineFragment != nil {
-				combinedSelections += selection.inlineFragment?.selectionSet ?? []
-			} else if selection.fragmentSpread != nil {
-				combinedSelections.append(selection)
-			}
-		}
-		
-		for selection in combinedSelections {
-			if let field = selection.field {
+			if let inlineFragment = selection.inlineFragment {
 				var render = "\n\(spacing)  {"
+				render.append("\n\(spacing)    selectionType: \"inlineFragment\",")
+				render.append("\n\(spacing)    typeCondition: \(renderTypeScriptTypeCondition(inlineFragment.typeCondition)),")
+				
+				if let renderedDirective = renderTypeScriptConditionalDirective(inlineFragment.conditionalDirective) {
+					render.append("\n\(spacing)    directive: \(renderedDirective),")
+				}
+				
+				let nextSpacingAmount = spacingAmount + 4
+				
+				if let renderedSelections = renderTypeScriptSelections(inlineFragment.selectionSet, args: [nil, nextSpacingAmount]) {
+					render.append("\n\(spacing)    selections: \(renderedSelections)")
+				}
+
+				render.append("\n\(spacing)  }")
+				
+				fieldRenders.append(render)
+				
+			} else if let fragmentSpread = selection.fragmentSpread {
+				var render = "\n\(spacing)  {"
+				render.append("\n\(spacing)    selectionType: \"fragmentSpread\",")
+				render.append("\n\(spacing)    name: \"\(fragmentSpread.name)\",")
+				render.append("\n\(spacing)    typeCondition: \(fragmentSpread.name.lowercasedFirstLetter)TypeCondition,")
+				
+				
+				if let renderedDirective = renderTypeScriptConditionalDirective(fragmentSpread.conditionalDirective) {
+					render.append("\n\(spacing)    directive: \(renderedDirective),")
+				}
+				
+				render.append("\n\(spacing)    selections: \(fragmentSpread.name.lowercasedFirstLetter)Selections")
+				
+				render.append("\n\(spacing)  }")
+				
+				fieldRenders.append(render)
+			} else if let field = selection.field {
+				var render = "\n\(spacing)  {"
+				render.append("\n\(spacing)    selectionType: \"field\",")
```

Comment:
Thoughts on including the `selectionType` on the standard field selection? We could leave this out and detect fields by checking if `selectionType` is `undefined`. The benefit of doing so is reducing the amount of data we send over the bridge when triggering queries. It feels kind of hacky thought which is why I opted to be explicit here.

Reply:
I think it'd be more straightforward to be explicit 👍🏻